content/posts/0020-manually-testing-changes-to-paketo-buildpacks.md

```diff
@@ -0,0 +1,249 @@
+---
+title: Manually testing changes to a Paketo Buildpack
+date: "2023-05-23"
+slug: manually-testing-changes-to-paketo-buildpacks
+author: anthonydahanne
+---
+
+*While this article is mainly about how to manually test Java family Paketo buildpacks, still, most of the instructions should work fine with other Paketo buildpacks; and even non Paketo buildpacks!*
+
+Suppose that you started working on a new feature for the [`spring-boot`](https://github.com/paketo-buildpacks/spring-boot/) buildpack...
+
+## Unit tests first 
+
+Of course, during your development, you will make sure that you did not break any tests, regularly running `go test -v ./...` but you would also add new tests covering your new code; the classic combination `go test -v ./... -coverprofile=coverage.out` followed with `go tool cover -html=coverage.out` will allow you to verify you did not forget to test some errors or edge cases.
+
+## Build and try the buildpack out before publishing it
+
+If the buildpack you are working on provides some integration tests (or end to end tests), I suggest you try and add some use cases to those tests.
+
+If not, then you probably want to try it out.
+
+### Build the Paketo Buildpack
+Most Paketo buildpacks provide a `scripts/build.sh` build script that you can use to build the `detect` and `build` binaries (luckily the required `buildpack.toml` is already at the root of the project).
+
+Just run the script and then check you have your buildpacks `bin` and `detect` executable files generated:
+
+```shell
+> scripts/build.sh
+> ls bin                                                                                                                                 
+build  detect helper main
+```
+
+### Mac arm64 note
+If you're using a Mac arm64 (M1, M2, etc.), you may need to update `scripts/build.sh` to precise you want to build for `amd64` (it will help if you publish your image and then try to run it on a  Linux amd64 host); to do that, just prepend `GOARCH="amd64"` before `go build`; like this:
```

Comment:
```suggestion
If you're using a Mac arm64 (M1, M2, etc.), you may need to update `scripts/build.sh` to specify that you want to build for `amd64` (it will help if you publish your image and then try to run it on a  Linux amd64 host); to do that, just prepend `GOARCH="amd64"` before `go build`; like this:
```

Reply:
oh nice, I did not think of that, much easier since I would not need to stash that change everywhere!